src/api/userApi.ts

```diff
@@ -0,0 +1,22 @@
+import HTTPTransport from "../utils/HTTPTransport";
+import { RegData, PassData } from "../type/interfaces";
+
+export default class UserApi {
+    protected http: HTTPTransport;
+
+    constructor() {
+        this.http = new HTTPTransport('/user');
+    };
+
+    public changeProfile(data: RegData, headers: object) {
+        return this.http.put('/profile', {data: data, headers: headers});
+    };
```

Comment:
http.put возвращает Promise, и changeProfile тоже возвращает Promise и вызывается через await. давай пометим как async
```suggestion
    public async changeProfile(data: RegData, headers: object) {
        return this.http.put('/profile', {data: data, headers: headers});
    };
```

То же самое следует сделать со всеми остальными методами api